heft-plugins/heft-api-extractor-plugin/src/ApiExtractorPlugin.ts

```diff
@@ -0,0 +1,246 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import type * as TApiExtractor from '@microsoft/api-extractor';
+import type {
+  IHeftTaskPlugin,
+  IHeftTaskRunHookOptions,
+  IHeftTaskSession,
+  HeftConfiguration,
+  IHeftTaskCleanHookOptions
+} from '@rushstack/heft';
+import { ConfigurationFile } from '@rushstack/heft-config-file';
+
+import { ApiExtractorRunner } from './ApiExtractorRunner';
+
+const PLUGIN_NAME: string = 'ApiExtractorPlugin';
+const PLUGIN_SCHEMA_PATH: string = path.resolve(__dirname, 'schemas', 'api-extractor-task.schema.json');
+const CONFIG_FILE_LOCATION: string = './config/api-extractor.json';
```

Comment:
Does this get exported from API Extractor? We should be able to do a typecheck-time validation that this is correct.